File: static/js/create.js
document.getElementById('createForm').addEventListener('submit', function (e) {
    e.preventDefault();

    // Corriger l'ID de l'input
    const nom = document.getElementById('itemName').value;

    fetch('/api/item', {
        method: 'POST',
        headers: { 'Content-Type': 'application/json' },
        body: JSON.stringify({ nom: nom })
    })
        .then(response => response.json())
        .then(data => {
            alert(data.message); // Affiche un message d'alerte avec le retour de l'API
            if (data.status === 'success') {
                document.getElementById('itemName').value = ''; // Réinitialiser le champ
                location.reload(); // Recharger la page pour actualiser la liste
            }
        })
        .catch(error => {
            console.error('Erreur lors de la création de l\'item :', error);
        });
});

File: blueprints/create.py
from flask import Blueprint, request, jsonify
from python.create import create_item

create_bp = Blueprint('create_bp', __name__)

@create_bp.route('/api/item', methods=['POST'])
def create():
    data = request.get_json()
    nom = data.get('nom')

    if not nom:
        return jsonify({'status': 'error', 'message': 'Nom requis'}), 400

    try:
        create_item(nom)
        return jsonify({'status': 'success', 'message': 'Item créé avec succès'}), 201
    except Exception as e:
        return jsonify({'status': 'error', 'message': str(e)}), 500
